command/agent/cache/api_proxy.go

```diff
@@ -47,7 +47,8 @@ func (ap *APIProxy) Send(ctx context.Context, req *SendRequest) (*SendResponse,
 	ap.logger.Info("forwarding request", "path", req.Request.URL.Path, "method", req.Request.Method)
 	resp, err := client.RawRequestWithContext(ctx, fwReq)
 	if err != nil {
-		return nil, err
+		// Bubble back the api.Response as well for error checking/handling at the handler layer.
+		return &SendResponse{Response: resp}, err
```

Comment:
Can `resp` be `nil` here?

Reply:
It can't be. If we don't return resp in here, the error check in handler wonder trigger, and thus we will be effectively always wrapping the error as an internal error. It's similar to the reasoning behind returning the response along with the error in `RawRequestWithContext` https://github.com/hashicorp/vault/blob/master/api/client.go#L796